sentinel-ai-configured-agents/src/main/java/com/phonepe/sentinelai/configuredagents/ConfiguredAgentFactory.java

```diff
@@ -47,6 +49,8 @@ public ConfiguredAgentFactory(
                 .orElseThrow(() -> new IllegalArgumentException("No MCP tool box found for upstream: " + upstream)))
                                  : null;
         this.customToolBox = customToolBox;
+        this.messagesPreProcessors = Optional.ofNullable(messagesPreProcessors)
```

Comment:
Can you not create optionals everywhere and use Objects.requireNotNull..  functions?